src/app.ts

```diff
@@ -1,124 +1,70 @@
-  const start: any = new Date();
-  res.on('close', () => {
-    const close: any = new Date();
-    const executionTime: any = close - start;
-    console.log(
-      `${req.method} ${req.originalUrl} execution time: ${executionTime}ms`
-    );
-  });
-
-  next();
-});
-
-app.get('/', (_, res: express.Response) => {
-  res.send('Welcome to hell!');
-});
-
-app.use('/api/users', userRouter);
-app.use('/api/groups', groupRouter);
-
-app.use(
-  (
-    err: Error,
-    req: express.Request,
-    res: express.Response,
-    next: express.NextFunction
-  ) => {
-    logger.error(err.stack || '');
-    res.status(500).send('Something broke!');
+import cors from 'cors';
+import helmet from 'helmet';
+import {
+  requestLoging,
+  executionTimeLoging,
+  serverErrorLoging,
+  verifyJWTToken
+} from './middlewares';
+import { sequelizer } from './data-access/sequelizer';
+import { authRouter, userRouter, groupRouter } from './routers';
+import './utils/passport';
+
+class Server {
+  public app: Application;
+
+  constructor() {
+    this.app = express();
+    this.config();
+    this.customMiddlewares();
+    sequelizer();
+    this.routes();
   }
-);
 
-(async () => {
-  try {
-    await sequelize.sync({
-      logging: console.log,
-      force: true
-    });
-    await User.bulkCreate(_USERS);
-    await Group.bulkCreate(_GROUPS);
+  public config(): void {
+    const { app } = this;
+    app.set('port', process.env.PORT || 3000);
+    app.set('jwtPrivateKey', process.env.JWT_PRIVATE_KEY || 'some_private_key');
+    app.use(bodyParser.urlencoded({ extended: true }));
+    app.use(bodyParser.json());
+    app.use(bodyParser.raw());
+    app.use(cors());
+    app.use(helmet());
+  }
 
-    User.belongsToMany(Group, {
-      as: 'Group',
-      through: 'UserGroup',
-      foreignKey: 'userId'
-    });
-    Group.belongsToMany(User, {
-      as: 'User',
-      through: 'UserGroup',
-      foreignKey: 'groupId'
-    });
-    User.hasMany(UserGroup, {
-      sourceKey: 'id',
-      foreignKey: 'userId',
-      as: 'groups'
-    });
-    Group.hasMany(UserGroup, {
-      sourceKey: 'id',
-      foreignKey: 'groupId',
-      as: 'users'
-    });
+  private customMiddlewares(): void {
+    const { app } = this;
+    app.use(requestLoging);
+    app.use(executionTimeLoging);
+    app.use(serverErrorLoging);
+    process
+      .on('unhandledRejection', (reason, p) => {
+        console.error(reason, 'Unhandled Rejection at Promise', p);
+      })
+      .on('uncaughtException', err => {
+        console.error(err, 'Uncaught Exception thrown');
+        process.exit(1);
+      });
+  }
 
-    const user = await User.create({
-      login: 'King',
-      password: '12345678',
-      age: 55,
-      isDeleted: false
-    });
+  public routes(): void {
+    const { app } = this;
+    app.get('/', (_, res: Response) => res.send('Welcome to hell!'));
+    app.use('/api/auth', authRouter);
+    app.use('/api/users', verifyJWTToken, userRouter);
```

Comment:
I guess it's better verify token globally. In case verification is not needed for certain path you can call next() immediately.